src/components/booklist.js

```diff
@@ -1,21 +1,47 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { removeBook } from '../redux/books/booksSlice';
+import { fetchBooks, removeBook } from '../redux/books/booksSlice';
 import Book from './book';
 import Input from './input';
 
 const Booklist = () => {
-  const books = useSelector((state) => state.books);
+  const books = useSelector((state) => state.books.books);
+  const isLoading = useSelector((state) => state.books.isLoading);
   const dispatch = useDispatch();
 
-  const deleteHandler = (id) => {
-    dispatch(removeBook({ id }));
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
+
+  const deleteHandler = async (id) => {
+    try {
+      await dispatch(removeBook(id));
+      dispatch(fetchBooks());
+      return id;
+    } catch (error) {
+      return error;
+    }
   };
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (!Array.isArray(books)) {
+    return <p>No books available.</p>;
+  }
 
   return (
     <>
       {books.map((book) => (
-        <Book key={book.id} book={book} deleteBook={deleteHandler} />
+        <div key={book.id}>
+          <Book
+            key={book.id}
+            title={book.title}
+            author={book.author}
+            category={book.category}
+            deleteBook={deleteHandler}
+          />
+        </div>
```

Comment:
- [ ] You have done great so far. :+1: :100: But looking at the video below, you can see that i get errors in the browser when I try running the application. The error points towards the `title` property not being present, I'll suggest you fix this. :+1: :100:  

- - Video reference:

https://github.com/tajulafreen/Book-Store/assets/107632780/b0472039-035f-424d-84f4-67802ef980ab